server/py/services/api/initial_data.py

```diff
@@ -993,6 +999,52 @@ def _create_project_summaries(db, db_session):
     db._upsert(db_session, project_summaries, ignore=True)
 
 
+def _init_system_id(db_session: sqlalchemy.orm.Session):
+    """
+    Initializes a system id for MLRun deployment.
+    The system id is first checked in the database. If it does not exist, the function checks if an id was set in the
+    config, and if neither is found, a new random one is generated and stored.
+    """
+
+    db = framework.db.sqldb.db.SQLDB()
+
+    # check if a system id already exists in the database
+    system_id = db.get_system_id(db_session)
+
+    if system_id is not None:
+        logger.debug("Existing system id found in the database", system_id=system_id)
+        mlrun.mlconf.system_id = system_id
+        return
+
+    logger.debug("System id not found in DB")
+    # check if the system id is already set in the config
+    system_id = _get_configured_system_id()
+
+    if system_id:
+        logger.debug("Using configured system id", system_id=system_id)
+    else:
+        # if no system id is found, generate a new one
+        system_id = _generate_system_id()
+    db.store_system_id(db_session, system_id)
+
+    # set the system id in mlrun config
+    mlrun.mlconf.system_id = system_id
+
+    logger.info("Initialized system ID", system_id=system_id)
+
+
+def _get_configured_system_id() -> typing.Optional[str]:
+    return mlrun.mlconf.system_id or None
+
+
+def _generate_system_id() -> str:
+    # Generate a random 32-bit unsigned integer, convert it to 4 bytes, and encode as a Base64 string without padding
+    random_number = random.getrandbits(32)
+    random_bytes = random_number.to_bytes(4, "big")
+    base64_str = base64.b64encode(random_bytes).decode("utf-8").rstrip("=")
+    return base64_str
```

Comment:
instead of using base64 which yields unexpected (or undesired) characters, simply use random character from list of [a-z|A-Z] and numbers. perhaps you can generate and iterate until you find a match that satisfies k8s dns name regex - as we want the system id to later on translate to k8s resource names / other usages that might enforce naming constrains.

usage example - `"".join(random.choices("0123456789" + string.ascii_lowercase, k=N))`

then enforce it matching regex of https://github.com/mlrun/mlrun/blob/196234944cd6d95eeaa81e9654a099ff8a4638d1/mlrun/utils/regex.py#L26

Reply:
@liranbg is right that base64 indeed has some issues with compatibility with DNS names. Another option would be to use the `altchars` parameter to replace `+` and `/` chars - see https://docs.python.org/3/library/base64.html#base64.b64encode , this parameter is actually there for this specific use-case. Usual replacements are to use `-` and `_` for this (but can use `.` as well).

Reply:
I used base64.urlsafe_b64encode to avoid issues with characters like + and / in DNS and Kubernetes names. Although the chance of starting with `-` or `_` is low, I included a check to further ensure the compatibility with the naming conventions.